pkg/cmd/roachtest/test_runner.go

```diff
@@ -1563,6 +1585,10 @@ func (r *testRunner) teardownTest(
 		getGoCoverArtifacts(ctx, c, t)
 	}
 
+	// Terminate tasks to ensure that any stray tasks are cleaned up.
```

Comment:
Question: if the test is successful, do we expect any tasks to not be cancelled already? If not maybe we should log when `Terminate` cleans up a task. I'm thinking it could be a warning the test writer maybe isn't properly cancelling their goroutines?

Reply:
Yes, I guess it depends on how the test implements it. A warning might be worthwhile, I'll see if I can add one.

Reply:
This is a little more involved than I anticipated since there is no easy way to tell if a group is still busy or have active tasks, so this would require more involved task tracking. Currently it has a mechanism that will complain about tasks that do not terminate within 5 minutes, whether the test failed or not. I think the idea to log 'leaky' tasks is still worthwhile, but maybe in a follow-up PR?